### ISSUE TITLE
Logs overwritten when restarting training from checkpoint.

### ISSUE THREAD

**Issue by alexcoca** [pytorch/fairseq#3334]
## 🐛 Bug

When restarting model from a checkpoint, any logs already existing are overwritten. 

### To Reproduce

1. Train a model for an epoch or until it saves a checkpoint.

2. Stop the training and start the training again. 

3. Observe the log gets overwritten. 

#### Code sample

Here is an excerpt of a log after restarting from checkpoint. Note the first epoch mentioned is `006` so the info about the first 5 epochs has been overwritten.

`| model transformer_wmt_en_de_big, criterion LabelSmoothedCrossEntropyCriterion
| num. model params: 196080640 (num. trained: 196080640)
| training on 4 GPUs
| max tokens per GPU = 3584 and max sentences per GPU = None
| loaded checkpoint /home/ac2123/rds/fairseq/examples/backtranslation/checkpoints/decoder_5_random/checkpoint_last.pt (epoch 5 @ 14670 updates)
| loading train data for epoch 5
| loaded 5187275 examples from: data-bin/wmt18_en_de/train.en-de.en
| loaded 5187275 examples from: data-bin/wmt18_en_de/train.en-de.de
| data-bin/wmt18_en_de train en-de 5187275 examples
| epoch 006:   1000 / 2934 loss=4.144, nll_loss=2.493, ppl=5.63, wps=20122, ups=0, wpb=52204.242, bsz=1757.914, num_updates=15671, lr=0.000505221, gnorm=0.324, clip=0.000, oom=0.000, wall=2661, train_wall=40119
| epoch 006:   2000 / 2934 loss=4.138, nll_loss=2.487, ppl=5.61, wps=20122, ups=0, wpb=52260.440, bsz=1765.725, num_updates=16671, lr=0.000489834, gnorm=0.323, clip=0.000, oom=0.000, wall=5261, train_wall=42680
| epoch 006 | loss 4.135 | nll_loss 2.485 | ppl 5.6 | wps 20107 | ups 0 | wpb 52242.230 | bsz 1767.987 | num_updates 17604 | lr 0.000476677 | gnorm 0.326 | clip 0.000 | oom 0.000 | wall 7687 | train_wall 45065
`

### Expected behavior

Have an option to prevent the log being overridden when restarting from checkpoint (something like `--reset-log`)

### Environment

 - fairseq 0.9
### Additional context

One may use spot instances on cloud providers or have limited running time on university clusters. Restarting from a previous checkpoint is necessary to continue training. It is desirable that the entire log is available at the end of training, not just the log after the last restart.
